### PR TITLE
Archive some artifacts as part of CI

### PR DESCRIPTION
**Pull request by ellismg**
This should teach the CI system archive the artifacts we care about as part of a PR.

### CODE REVIEW

**Comment by ellismg** [comment]
/cc @weshaggard @ericstj 

I'll merge this in tonight and do a dummy PR to make sure the jobs work as expected.  Note that CI does not do per push jobs (but we could fix that here if we wanted) so we'll have to pull the artifacts off of Jenkins for a dummy PR).

File: netci.groovy [hunk 1]
```diff
@@ -61,12 +62,20 @@ platformList.each { platform ->
...
+
     if (!(os == 'Windows_NT' && architecture == 'arm')) {
         Utilities.addXUnitDotNETResults(newJob, '**/*-testResults.xml')
     }
-    
+
     Utilities.addGithubPRTriggerForBranch(newJob, branch, "${os} ${architecture} ${configuration} Build")
+
+    ArchivalSettings settings = new ArchivalSettings();
+    def archiveString = ["tar.gz", "zip", "deb", "msi", "nupkg", "exe"].collect { "artifacts/*/packages/*.${it},artifacts/*/corehost/*.${it}" }.join(",")
```
Comment (ericstj) [inline netci.groovy]: .pkg as well for OS X?

Comment (ellismg) [inline netci.groovy]: Good catch.  Will do.

**Comment by ellismg** [comment]
@dotnet-bot test ci please

**Pull request merged**